proxy_test.go

```diff
@@ -98,29 +104,45 @@ func TestReverseProxyHash(t *testing.T) {
 		assert.True(t, numHits > 0, "One down - some hit")
 		assert.True(t, numMisses < n, "One down - not all missed")
 		assert.True(t, numMisses > 0, "One down - some missed")
+		assert.Equal(t, n, numHits+numMisses, "One down - hits and misses adds up")
 		accHits += numHits
 		accMisses += numMisses
 	}
 	assert.Equal(t, n, accMisses, "Misses add up")
 }
 
 func measHitMiss(t *testing.T, client *redis.Client, n int, keyTempl string) (numHits, numMisses int, err error) {
+	type tresult struct {
+		hit bool
+		err error
+	}
+	results := make(chan tresult)
 	for i := 0; i < n; i++ {
 		key := fmt.Sprintf(keyTempl, i)
-		ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
-		defer cancel()
-		args := client.Query(ctx, "GET", key)
-		for j := 0; j < args.Len(); j++ {
+		go func(key string, rq chan<- tresult) {
+			ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+			defer cancel()
+			args := client.Query(ctx, "GET", key)
+			assert.Equal(t, 1, args.Len(), "GET returns a single value")
 			var v string
 			args.Next(&v)
-			if v == "1" {
-				numHits++
+			hit := v == "1"
+			if err := args.Close(); err != nil {
+				rq <- tresult{false, err}
 			} else {
-				numMisses++
+				rq <- tresult{hit, nil}
```

Comment:
Styling tip here, `false` and `nil` often don't communicate well to the reader what they represent, when possible using named fields helps better communicate the intent, for example here you could change the code to be:
```go
if err := args.Close(); err != nil {
    rq <- tresult{err: err}
} else {
    rq <- tresult{hit: hit}
}
```
That way you leverage zero-values to have your object properly initialize while helping the reader focus on the important part of the code.